test/kitchen/tasks/unit-tests/kitchen_unit_tests.py

```diff
@@ -0,0 +1,35 @@
+import unittest
+
+from invoke import Exit, MockContext
+
+from ..kitchen import should_rerun_failed
+
+
+class TestKitchenInvokeMethod(unittest.TestCase):
+    def test_no_rerun_gotest(self):
+        mock_context = MockContext()
+        with self.assertRaises(Exit):
+            should_rerun_failed(mock_context, "tasks/unit-tests/gotest-failed-runlog")
+
+    def test_rerun_gotest(self):
```

Comment:
I am not sure it would make the test more efficient, when the function is called what will determine if the test should be rerun or not is the exit code. If the exit code is 0 we consider the test should be rerun otherwise we do not rerun the tests. The print here is just to give some information to the person reading the logs in the CI jobs.

Reply:
Agreed, it's just you don't assert nothing in this test if I'm correct?

Reply:
Hmm yes you are right I think, if the `should_rerun_failed` function return an error it fails with `self.fail` otherwise the test is successful. Does the test explicitly need to assert something?